# Samples

![ss](https://cdn.upgrad.com/UpGrad/temp/687eade2-b688-4d44-b170-ea8460b187e0/Table%201.png)

Given
------

For an upcoming government project, you want to find the average height of the students in Class VIII of a given school. Instead of asking each student, suppose you took a few students as your sample and wrote the data down:

<br>

Roll Number | Height
---- | ----
8012	| 121.92 cm
8045 |	133.21 cm
8053	| 141.34 cm
8099	| 126.23 cm
8125 |	175.74 cm


In [1]:
rn = {
  8012 : 121.92,
  8045 : 133.21,
  8053 : 141.34,
  8099 : 126.23,
  8125 : 175.74
}

rn

{8012: 121.92, 8045: 133.21, 8053: 141.34, 8099: 126.23, 8125: 175.74}

School Kids Sample
---------

What is the sample size for this sample?

    5

    40334

    698.44

    10

-------------

    5

    ✓ Correct
    Feedback:

In this sample, you have data on 5 students.

Hence, the sample size is equal to 5.

In [2]:
len(rn)

5

School Kids Sample
------------

What is the sample mean (
¯
X
)?

    133.21

    139.69

    146.76

    141.34

--------

    139.69

    ✓ Correct
    Feedback:

The sample mean (
¯
X
)

= (121.92 + 133.21 + 141.34 + 126.23 + 175.74) / 5

    = 139.69

In [3]:
rnv = rn.values()
avg = sum(rnv) / len(rnv)

round(avg, 2)

139.69

School Kids Sample
-------

What will be the sample's standard deviation (S)?

    22.84

    23.67

    19.19

    21.45

--------

    21.45

    ✓ Correct
    Feedback:

We found in the previous question that the sample mean (
¯
X
) = 139.69.

Now, we can find
∑
(
X
i
−
¯
X
)
2
, which turns out to be 1841.26.

Dividing this by n-1, i.e, 4, we get the value of S^2 as 460.315

The value of S, thus, will be equal to √(460.315) = 21.45


In [4]:
import numpy as np

rnv = list(rn.values())
rnv = np.array(rnv)
rnv

array([121.92, 133.21, 141.34, 126.23, 175.74])

In [5]:
avg = sum(rnv) / len(rnv)
avg

139.68800000000002

In [6]:
diff = (rnv - avg) ** 2
diff

array([ 315.701824,   41.964484,    2.729104,  181.117764, 1299.746704])

In [7]:
S = np.sum(diff) / (len(rn) - 1)
S

460.31497000000013

In [8]:
np.sqrt(S)

21.454952109012037

In [9]:
def stdn_sample(d):
  import numpy as np
  x = list(d.values())
  rnv = np.array(x)
  avg = sum(rnv) / len(rnv)
  diff = (rnv - avg) ** 2
  S = np.sum(diff) / (len(d) - 1)
  return np.sqrt(S)

In [10]:
# Ans.

round(stdn_sample(rn), 2)

21.45

# Sampling Distributions

The sampling distribution’s mean is denoted by μX, as it is the mean of the sampling means.

Let’s see what it is for this sampling distribution.

Sampling Distribution
----------

What is μX for our sampling distribution?

(The data for the 100 sample means is given in the CSV file upgrad_samples.csv).

    2.812

    2.014

    2.348

    2.686

-----------

    2.348

    ✓ Correct
    Feedback:

The sampling distribution’s mean would be equal to the sum of all the sample means, divided by 100.

This comes out to be 2.348

In [11]:
import pandas as pd

df = pd.read_csv('Inferntial Statistics - UpGrad Samples.csv')
df.head()

Sample No.  Sample Mean
0           1          3.2
1           2          2.6
2           3          2.8
3           4          2.0
4           5          3.0

In [12]:
df.describe()

Sample No.  Sample Mean
count  100.000000   100.000000
mean    50.500000     2.348000
std     29.011492     0.426989
min      1.000000     1.400000
25%     25.750000     2.000000
50%     50.500000     2.400000
75%     75.250000     2.600000
max    100.000000     3.200000

In [13]:
df.mean()

Sample No.     50.500
Sample Mean     2.348
dtype: float64

In [14]:
df['Sample Mean'].mean()

2.3480000000000003

In [15]:
mn = df.loc[:, 'Sample Mean'].mean()
mn

2.3480000000000003

In [16]:
# Ans.

round(mn, 3)

2.348

Sampling Distribution
-----------

What is the standard deviation of our sampling distribution?

(The data for the 100 sample means is given in the CSV file upgrad_samples.csv)

    0.4248

    0.4270

    0.4512

    0.3986

------------

    0.4248

    ✓ Correct
    Feedback:

Using the mean from the previous question, you can find the standard deviation using the formula -

(
S
t
a
n
d
a
r
d
D
e
v
i
a
t
i
o
n
)
** 2
=
∑
(
X
i
−
¯
X
)
** 2
/ n

Note that you would `divide by n and not n-1` as you have the data for all 100 entries of the distribution and you don't need to sample the distribution.

Now, using the formula, you get that the value of the standard deviation of the distribution is 0.4248

In [17]:
def stdn_un_sample(d):
  import numpy as np
  x = list(d.values())
  rnv = np.array(x)
  avg = sum(rnv) / len(rnv)
  diff = (rnv - avg) ** 2
  S = np.sum(diff) / len(d)
  return np.sqrt(S)

In [18]:
dv = dict(df.values)
s = stdn_un_sample(dv)
s

0.4248482081873478

In [19]:
# Ans.

round(s, 4)

0.4248

# Properties of Sampling Distributions

![ss](https://cdn.upgrad.com/UpGrad/temp/02ee8693-3687-4a08-863d-39f237f55658/Table%202.png)

# Central Limit Theorem

So, there are two important properties of a sampling distribution of the mean:

- Sampling distribution’s mean (
μ
X
) = Population mean (μ)

- Sampling distribution’s standard deviation (Standard error) =
σ /
√
n
, where σ is the population’s standard deviation and n is the sample size

- For n > 30, the sampling distribution becomes a normal distribution.

# Practice Questions - Part I

Let’s say that you work for a news agency, which is conducting an exit poll for the MCD (Municipal Corporation of Delhi) elections.

You have been tasked with predicting the winner for ward 75N (Ashok Vihar).

You ask 100 randomly selected voters from this ward to name the party they had voted for.

The data thus collected is as given in the following table.

<br>
<center>

Contesting Party | Number of Voters
--- | ---
BJP | 58
INC | 42

</center>
<br>

From this sample, you have to estimate the percentage of voters that may have voted for BJP.

So, you define X as the proportion of people that voted for BJP. Then, the frequency distribution for X would be as follows:

<br>
<center>

X | Frequency
--- | ---
1 | 58
0 | 42

</center>
<br>

Now, you have to find the mean for X, which is equal to

(0 + 0 + 0 + ... 42 times) + (1 + 1 + 1 + ... 58 times),

divided by the total frequency, i.e., 100. So, the

-----------

`mean` = $\frac{(0*42)+(1*58)}{100}$ = 0.58, or 58%

----------

Also, you would have to find the standard deviation for this sample of 100 voters.

Since the mean is 0.58, the sample’s

-----------

`variance` = $\frac{((0-0.58)^{2}*42)+((1-0.58)^{2}*58)}{100-1}$ = 0.2461.

------------

So, the standard deviation is equal to its square root,

    i.e., 0.496, or 49.6%

In [20]:
mn = (0 * 42 + 1 * 58) / 100
mn, f'{round(mn * 100, 0)} %'

(0.58, '58.0 %')

In [21]:
var = (((0 - 0.58)**2 * 42) + ((1 - 0.58)**2 * 58)) / (100 - 1)
var

0.24606060606060606

In [22]:
std = var ** (1/2)
std, f'{round(std * 100, 1)} %'

(0.4960449637488582, '49.6 %')

Voter Sample
-----------

Now, let’s say you define Y as the proportion of people that voted for the INC party.

Clearly, the mean of Y will be 0.42.

What would be the standard deviation of Y?

    24.61%

    49.6%

    50.4%

    75.39%

-----------

    49.6%

    ✓ Correct
    Feedback:

There would be 42 people for whom Y would be equal to 1, and 58 people for whom Y would be equal to 0.

Hence, the sample’s

`variance` = $\frac{((1-0.42)^{2}*42)+((0-0.42)^{2}*58)}{100-1}$ = 0.2461.

<br>
<img src="https://latex.codecogs.com/svg.image?\inline&space;\huge&space;\frac{((1-0.42)^{2}*42)&plus;((0-0.42)^{2}*58)}{100-1}" title="https://latex.codecogs.com/svg.image?\inline \huge \frac{((1-0.42)^{2}*42)+((0-0.42)^{2}*58)}{100-1}" />

So, the standard deviation is equal to its square root,

    i.e. 0.496 or 49.6%

In [23]:
var = (((1 - 0.42)**2 * 42) + ((0 - 0.42)**2 * 58)) / (100 - 1)
var

0.24606060606060606

In [24]:
# Ans.

f'{round(var ** (1/2) * 100, 1)} %'

'49.6 %'

Voter Sample
---------

Let’s say you actually took some more samples, each of size 100, and made a sampling distribution for X, the proportion of people that voted for BJP.

The mean of this sampling distribution is μX = 0.50 and the standard error = 0.052

So, for the whole survey population, i.e. all the voters in ward 75N, the proportion of people that voted for BJP is approximately equal to:

    50%

    55.8%

    44.2%

    10.2%

---------

    50%

    ✓ Correct
    Feedback:

You have to find the mean for all the people or, in other words, the population mean.

Using CLT, you can say that the mean of the sampling distribution = the population mean.

As the question says that the sampling distribution’s mean = 0.5, the population mean also = 0.5 or 50%

Voter Sample
-------------

Similarly, let’s say you made a sampling distribution for Y, the proportion of people that voted for INC.

The mean of this sampling distribution is μX = 0.50 and the standard error is equal to 0.048.

Now, for the whole survey population, i.e. all the voters in ward 75N, the standard deviation of Y is equal to:

50%

4.8%

48%

10.4%

--------

    48%

    ✓ Correct
    Feedback:

You have to find the standard deviation for all people or, in other words, the population standard deviation σ.

Using CLT, you can say that `(σ/√n) = SE`.

Hence, (σ/√100) = 0.048, which gives σ = .48 or 48%

In [25]:
se = 0.048
n = 100

std = se * (n ** 0.5)
std

0.48

In [26]:
f'{round(std * 100, 0)} %'

'48.0 %'

# [Estimating Mean Using CLT](https://live-session-recording-service.s3.ap-south-1.amazonaws.com/production/173111/cl9qkCNpmrCHiFhP5RV869acqbX55y6S.mp4)

You are doing sampling because you want to find the population mean, albeit in the form of an interval. The three steps to follow are as follows:

First, take a sample of size n.

Then, find the mean X and standard deviation S of this sample.

Now, you can say that for a y% confidence level, the confidence interval for the population mean $\mu$ is given by

<center>

($\bar{X}-\frac{Z^{*}S}{\sqrt{n}}, \bar{X}+\frac{Z^{*}S}{\sqrt{n}}$)

<br>

-------------

### What is the difference between Z* and Z-score?

![ss](https://cdn.upgrad.com/UpGrad/temp/9b189b5e-bd87-4b3d-8b5d-e33609321b01/Capture19.PNG)

</center>

Z-score is the value associated with a particular cumulative probability whereas Z* is the Z-score associated with a particular confidence level.

`If someone asks, what is the Z-score associated with 0.95 or 95%  probability, you just have to look up the table and find the corresponding value associated with it`.

In this case, it would come up as  1.65 (or 1.645 to be more exact).

`However if someone asks you what is the Z* for 95% confidence level, then first you need to find the corresponding cumulative probability associated with that confidence level`.

This can be easily calculated with the following formula: if y is the confidence level in percentage then the probability is given as y/100 +(1-y/100)/2.

Thus for y=95, we have the probability as 95/100+(1-95/100)/2= 0.975.

And the Z-score associated with 0.975 probability is 1.96.

You can verify the rest of the values given above with this formula.

# Confidence Interval - Example

In [27]:
def con_int(n, x, s, z):
  ci = (z * s) / (n ** 0.5)
  return x - ci, x + ci

In [28]:
# Given
n = 100
x = 2.3
s = 0.3
c = 0.99

# z for c = 99% is 2.58
z = 2.576

con_int(n, x, s, z)

(2.22272, 2.37728)

Mean Estimation Using CLT
---------

Recall that 50.5% of the 10,000 people surveyed preferred feature B to feature A.

So, if X = the proportion of people that prefer feature B to feature A, then, for this sample, $\bar{X}$ = 0.505 (50.5%) and n = 10,000.

In addition to this, you've been told that the sample’s standard deviation S = 0.2 (20%).

For the above case, the 95% confidence interval is:

    (50.11%, 50.89%)

    (50.50%, 50.89%)

    (50.11%, 50.50%)

    (50.30%, 50.69%)

----------

    (50.11%, 50.89%)

    ✓ Correct
    Feedback:

From the above, you have `X = 50.5%, S = 20% and n = 10,000`.

The confidence interval is given by ($\bar{X}-\frac{Z^{*}S}{\sqrt{n}}, \bar{X}+\frac{Z^{*}S}{\sqrt{n}}$), where Z* is the Z score associated with 95% confidence level = 1.96.

Hence, the 95% confidence interval would be 50.5 = (50.11% to 50.89%)

In [29]:
# Given
n = 10000
x = 0.505
s = 0.2
c = 0.95

# z for c = 95% is 1.96
z = 1.96

ci = con_int(n, x, s, z)
ci

(0.50108, 0.50892)

In [30]:
a = f'{round(ci[0] * 100, 2)} %'
b = f'{round(ci[1] * 100, 2)} %'

a, b

('50.11 %', '50.89 %')

In [31]:
f'{round(np.array(ci).mean() * 100, 1)} %'

'50.5 %'

Mean Estimation Using CLT
---------

For the above case, the 99% confidence interval is:

    (50.24%, 50.76%)

    (49.98%,51.02%)

    (50.06%, 50.94%)

    (49.82%, 51.18%)

-----------

    (49.98%,51.02%)

    ✓ Correct
    Feedback:

From the above, you have X= 50.5%, S = 20% and n = 10,000.

The confidence interval is given by ($\bar{X}-\frac{Z^{*}S}{\sqrt{n}}, \bar{X}+\frac{Z^{*}S}{\sqrt{n}}$), where Z* is the Z score associated with 99% confidence level = 2.58.

Hence, the 99% confidence interval would be 50.5 = (49.98% to 51.02%).

In [32]:
# Given
n = 10000
x = 0.505
s = 0.2
c = 0.99

# z for c = 99% is 2.58
z = 2.58

ci = con_int(n, x, s, z)
ci

(0.49984, 0.5101600000000001)

In [33]:
a = f'{round(ci[0] * 100, 2)} %'
b = f'{round(ci[1] * 100, 2)} %'

a, b

('49.98 %', '51.02 %')

In [34]:
f'{round(np.array(ci).mean() * 100, 1)} %'

'50.5 %'

Mean Estimation Using CLT
-----------

Now, you want to make the claim that more than 50% people prefer feature B over feature A.

Which of the following statement/s about this claim is/are true?

1. You can make this claim with 90% confidence

2. You can make this claim with 95% confidence

3. You can make this claim with 99% confidence


    All of these statements are correct

    Only statement 1 is correct

    Only Statement 1 and 2 are correct

-------------

    Only Statement 1 and 2 are correct

    ✓ Correct
    Feedback:

90% confidence interval for μ is (50.17%, 50.83%).

Hence, with 90% confidence, you can say that more than 50% people prefer B over A.

Similarly, since 95% confidence interval is (50.11%, 50.89%), you can say with 95% confidence that more than 50% people prefer B over A.

However, the 99% confidence interval is (49.98%, 51.02%).

Hence, at 99% confidence level, you cannot say for sure that more than 50% people prefer B over A.

In [35]:
# Given
n = 10000
x = 0.505
s = 0.2
c = 0.90

# z for c = 90% is 1.65
z = 1.65
ci = con_int(n, x, s, z)

a = f'{round(ci[0] * 100, 2)} %'
b = f'{round(ci[1] * 100, 2)} %'
c = f'{round(np.array(ci).mean() * 100, 1)} %'

a, b, c

('50.17 %', '50.83 %', '50.5 %')

# Summary: Central Limit Theorem - Part II

https://cdn.upgrad.com/UpGrad/temp/bb87fd1e-8472-4405-9d31-d6d8ba22d346/Lecture%20Notes%20-%20Inferential%20Statistics.pdf

# Practice Questions - Part II

Let’s say you work as a data analyst at a pharma company, which manufactures an antipyretic drug (in tablet form) with paracetamol as the active ingredient.

The amount of paracetamol specified by the drug regulatory authorities is 500 mg with an allowed error of 10%.

Anything below 450 mg would be a quality issue for your company, as the drug would become ineffective, while anything above 550 mg would lead to a serious regulatory issue.



There are 10 identical manufacturing lines in the plant, each of which produces approximately 10,000 tablets per hour.



You want to take some samples, measure the amount of paracetamol, and test if the manufacturing process is running successfully.

You have the resources and time to take a sample of 100 tablets and measure the paracetamol content in each.



For the 100 tablets that you took as a sample, you find that the mean paracetamol content is 530 mg and the standard deviation is 100 mg.


Now you want to know what the average content is for all the tablets in the plant.

You are thinking of reporting the average as a confidence interval, for which you are 95% confident.



With this information, answer the questions given below.

Paracetamol Content
---------

What is the MOE (margin of error) for 95% confidence level?

    9.8 mg

    19.6 mg

    39.2 mg

    49 mg

--------

    19.6 mg

    ✓ Correct
    Feedback:

If X is the defined as the paracetamol content, then for this sample of X, sample mean X = 530 mg, sample standard deviation S = 100 mg and sample size n = 100.

Also, for 95% confidence interval, Z* is 1.96.

Now, you know that the `margin of error`

=
Z
∗
S /
√
n

=
1.96
∗
100 /
√
100

= 19.6

In [36]:
def moe(n, s, z):
  return (z * s) / (n ** 0.5)

In [39]:
# Given
n = 100
s = 100
c = 0.95

# z for c = 95% is 1.96
z = 1.96

m = moe(n, s, z)
m

19.6

Paracetamol Content
----------

What is the confidence interval for 95% confidence level?

    510.4 to 530

    530 to 549.6

    510.4 to 549.6 ✓ Correct

    512.1 to 547.9

In [41]:
# Given
n = 100
x = 530
s = 100
c = 0.95

# z for c = 95% is 1.96
z = 1.96

ci = con_int(n, x, s, z)
ci

(510.4, 549.6)

Paracetamol Content
---------

What is the confidence interval for 90% confidence level?

    514.7 to 545.3

    511.3 to 548.7

    513.5 to 546.5

In [42]:
# Given
n = 100
x = 530
s = 100
c = 0.90

# z for c = 90% is 1.65
z = 1.65

ci = con_int(n, x, s, z)
ci

(513.5, 546.5)

Paracetamol Content
----------

Now, you claim that the tablet is fit to consume and effective, i.e., its Paracetamol content is less than 550 mg and more than 450 mg.

Can you make this claim at -

    95% confidence level?

    90% confidence level?

Why or why not? (Word Limit: 125)

-----------

At a 95% confidence level, we can make the claim because the wider range of acceptance allows for a higher level of certainty, whereas at a 90% confidence level, we cannot make the claim without a narrower range that provides a smaller margin of error.